src/containerapp/azext_containerapp/_utils.py

```diff
@@ -1506,42 +1506,19 @@ def is_registry_msi_system(identity):
 
 
 def validate_environment_location(cmd, location):
-    from ._constants import MAX_ENV_PER_LOCATION
-    from .custom import list_managed_environments
-    env_list = list_managed_environments(cmd)
-
-    locations = [loc["location"] for loc in env_list]
-    locations = list(set(locations))  # remove duplicates
-
-    location_count = {}
-    for loc in locations:
-        location_count[loc] = len([e for e in env_list if e["location"] == loc])  # pylint: disable=used-before-assignment
-
-    disallowed_locations = []
-    for _, value in enumerate(location_count):
-        if location_count[value] > MAX_ENV_PER_LOCATION - 1:
-            disallowed_locations.append(value)
-
     res_locations = list_environment_locations(cmd)
-    res_locations = [loc for loc in res_locations if loc not in disallowed_locations]
 
     allowed_locs = ", ".join(res_locations)
 
     if location:
         try:
             _ensure_location_allowed(cmd, location, CONTAINER_APPS_RP, "managedEnvironments")
+
+            return location
         except Exception as e:  # pylint: disable=broad-except
             raise ValidationError("You cannot create a Containerapp environment in location {}. List of eligible locations: {}.".format(location, allowed_locs)) from e
-
-    if len(res_locations) > 0:
-        if not location:
-            logger.warning("Creating environment on location %s.", res_locations[0])
-            return res_locations[0]
-        if location in disallowed_locations:
-            raise ValidationError("You have more than {} environments in location {}. List of eligible locations: {}.".format(MAX_ENV_PER_LOCATION, location, allowed_locs))
-        return location
     else:
-        raise ValidationError("You cannot create any more environments. Environments are limited to {} per location in a subscription. Please specify an existing environment using --environment.".format(MAX_ENV_PER_LOCATION))
+        return res_locations[0]
```

Comment:
add null check

Reply:
res_location will not be empty after the change as we don't filter by remaining quota anymore